File: design-pattern-java/app/src/test/java/design/pattern/java/patterns/creational/prototype/PrototypeTest.java
package design.pattern.java.patterns.creational.prototype;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class PrototypeTest {
  @Test
  public void shouldCopyValuesWhenCopyIsCalled() {
    Car car = new Car("test");
    Car carCopy = (Car)car.copy();
    Assertions.assertEquals(car.getName(), carCopy.getName());
  }
  @Test
  public void shouldHaveDiferenteReferenceWhenCopyIsCalled() {
    Car car = new Car("test");
    Car carCopy = (Car)car.copy();
    Assertions.assertNotEquals(car, carCopy);
  }
}

File: design-pattern-jvm/app/src/test/java/design/pattern/kotlin/patterns/creational/prototype/PrototypeTest.kt
package design.pattern.kotlin.patterns.creational.prototype

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class PrototypeTest {
    @Test
    fun shouldCopyValuesWhenCopyIsCalled() {
        val car = Car("test")
        val carCopy = car.copy() as Car
        Assertions.assertEquals(car.name, carCopy.name)
    }

    @Test
    fun shouldHaveDifferentReferenceWhenCopyIsCalled() {
        val car = Car("test")
        val carCopy = car.copy() as Car
        Assertions.assertNotEquals(car, carCopy)
    }
}